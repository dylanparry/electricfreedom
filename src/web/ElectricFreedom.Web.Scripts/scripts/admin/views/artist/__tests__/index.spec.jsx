import React from 'react';
import PropTypes from 'prop-types';
import createContext from 'react-router-test-context';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';

import ArtistSwitch from '../index';
import ArtistAdd from '../add';
import ArtistEdit from '../edit';
import ArtistList from '../list';

const setup = (pathname = '/') =>
{
  const options = {
    context: createContext({
      location: { pathname },
    }),
    childContextTypes: {
      router: PropTypes.object,
    },
  };
  const wrapper = mount(<ArtistSwitch />, options);

  return {
    wrapper,
    snapshot: toJson(wrapper),
  };
};

describe('ArtistSwitch', () =>
{
  test('render', () =>
  {
    const { snapshot } = setup();

    expect(snapshot).toMatchSnapshot();
  });

  test('route "/artists/add" renders ArtistAdd', () =>
  {
    const { wrapper } = setup('/artists/add');

    expect(wrapper.find(ArtistAdd).exists()).toBe(true);
  });

  test('route "/artists/:artistId" renders ArtistEdit', () =>
  {
    for (let i = 1; i <= 5; i += 1)
    {
      const { wrapper } = setup(`/artists/${i}`);

      const artistEdit = wrapper.find(ArtistEdit);
      expect(artistEdit.exists()).toBe(true);
      expect(artistEdit.props().match.params.artistId).toBe(i.toString());
    }
  });

  test('route "/artists" renders ArtistList', () =>
  {
    const { wrapper } = setup('/artists');

    expect(wrapper.find(ArtistList).exists()).toBe(true);
  });
});
