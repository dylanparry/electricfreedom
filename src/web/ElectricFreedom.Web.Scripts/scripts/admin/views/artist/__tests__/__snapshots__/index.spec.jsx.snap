// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistSwitch render 1`] = `
<artistSwitch>
  <Switch />
</artistSwitch>
`;
